addon/index.js

```diff
@@ -21,7 +22,8 @@ export default function localStorageDecorator(customLocalStorageKey) {
   return function (target, key, descriptor) {
     const localStorageKey = customLocalStorageKey ?? key;
 
-    if (!localStorageCache.has(localStorageKey)) {
+    if (!keyCache.has(localStorageKey)) {
```

Comment:
Some inline comment why we can not use `localStorageCache.has(localStorageKey)` might be helpful. Maybe like this:

```suggestion
    // Check if key is already managed. If it is not managed yet, initialize it
    // in localStorageCache with the current value in local storage.
    // Need to use a separate, not tracked data store to do this check
    // because a tracked value (`localStorageCache`) must not be read
    // before it is set.
    if (!keyCache.has(localStorageKey)) {
```